includes/class-newspack-newsletters.php

```diff
@@ -90,6 +101,170 @@ public function __construct() {
+	 * 
+	 * @return mixed
+	 */
+	public function npublic_columns( $column_array ) {
+
+		$column_array['public'] = 'Published as a post';
+
+		return $column_array;
+	}
+
+	/**
+	 * Populate column value
+	 * 
+	 * @param string $column_name Column name.
+	 * @param int    $post_id Post ID.
+	 */
+	public function populate_columns( $column_name, $post_id ) {
+
+		// if you have to populate more that one columns, use switch().
+		switch ( $column_name ) :
+			case 'public':
+				echo ( get_post_meta( $post_id, 'is_public', true ) == '1' ) ? 'Yes' : 'No';
+				break;
+		endswitch;
+
+	}
+
+	/**
+	 * Display Make Newsletters public checkbox
+	 * 
+	 * @param string $column_name Coulumn name.
+	 * @param string $post_type Post Type.
+	 */
+	public function quick_edit_fields( $column_name, $post_type ) {
+
+		switch ( $column_name ) :
+			case 'public':
+				wp_nonce_field( 'quick_edit_npublic_nonce', 'npublic_nonce' );
+				echo '<fieldset class="inline-edit-col-right"><div class="inline-edit-col"><div class="inline-edit-group wp-clearfix">';
+				echo '<label class="alignleft">
+						<input type="checkbox" name="npublic">
+						<span class="checkbox-title">Make newsletter page public?</span>
+					</label>';
+
+				// for the LAST column only - closing the fieldset element.
+				echo '</div></div></fieldset>';
+
+				break;
+		endswitch;
+
+	}
+
+	/**
+	 * Save value in public newsletters checkbox
+	 * 
+	 * @param int $post_id Post ID.
+	 */
+	public static function quick_edit_save( $post_id ) {
+
+		// check user capabilities.
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+
+		// check nonce.
+		if ( isset( $_POST['npublic_nonce'] ) ) {
+			$pnonce = sanitize_key( $_POST['npublic_nonce'] );
+		}
+		if ( ! empty( $pnonce ) ) {
+			if ( wp_verify_nonce( $pnonce, 'quick_edit_npublic_nonce' ) ) {
+
+				// update checkbox.
+				if ( isset( $_POST['npublic'] ) ) {
+					update_post_meta( $post_id, 'is_public', '1' );
+				} else {
+					update_post_meta( $post_id, 'is_public', '' );
+				}
+			}
+		}
+
+	}
+	/**
+	 * Register bulk action fields in bulk action dropdown
+	 * 
+	 * @param string $bulk_actions Bulk actions.
+	 * 
+	 * @return mixed 
+	 */
+	public static function register_bulk_actions( $bulk_actions ) {
+		$bulk_actions['public_newsletters']     = __( 'Make Newsletters Public', 'newspack_newsletters' );
+		$bulk_actions['non_public_newsletters'] = __( 'Make Newsletters Non-public', 'newspack_newsletters' );
+		return $bulk_actions;
+	}
+
+	/**
+	 * Bulk action handler onclick event
+	 * 
+	 * @param string $redirect_to Redirect URL.
+	 * @param string $doaction Action.
+	 * @param int    $post_ids Post IDs.
+	 */
+	public static function bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
+		$nonce = wp_create_nonce( 'npublic' );
+		if ( 'public_newsletters' == $doaction ) {
+			foreach ( $post_ids as $post_id ) {
+				update_post_meta( $post_id, 'is_public', 1 );
```

Comment:
I don't think the nonce in this function is going to be useful right now, since this function updates the post meta *before* checking a nonce.

Reply:
This nonce, I am passing at the time of admin url request. Hope this will be fine